### PR TITLE
better char detction

### PR DESCRIPTION
**Pull request by moritzschaefer**

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -21,8 +21,11 @@
 NUM_D = 200  # number of dictionary entries
 
 
-TEXT_RECOGNITION_THRESHOLD = 0.7
-BOUNDING_BOX_THRESHOLD =  32*32*0.6
+# the thresholds depend on STEP_SIZE
+STEP_SIZE = 4
+TEXT_RECOGNITION_THRESHOLD = 36*4/STEP_SIZE
```
Comment (shansmann) [inline src/config.py:26]: whats that Threshold for?

File: src/feature_extraction.py [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
     # get value |D^Tx| - ALPHA
-    tmp = np.abs(np.dot(dictionary.T, flattend_x)) - config.ALPHA
+    tmp = np.abs(np.dot(get_z.dictionary.T, flattend_x)) - config.ALPHA
 
     # choose maximum from 0 and tmp
     indexes = np.where(tmp < float(0))
     tmp[indexes] = 0
 
     return tmp
+get_z.dictionary = None
```
Comment (shansmann) [inline src/feature_extraction.py:30]: looks like a class thing, never seen that on a function - did u test this?

File: src/predict_test_img.py [hunk 3]
```diff
@@ -77,20 +83,18 @@ def predict_images():
 
     for filename in image_files:
         img = cv2.imread(filename)
-        predicted_layers = get_prediction_values(img, text_model)
+        predicted_layers = get_prediction_values(img, text_model, step_size)
         for layer_img, layer_predictions in predicted_layers:
             # compute
+            cv2.imshow("image 1", layer_img)
```
Comment (shansmann) [inline src/predict_test_img.py:89]: i'd save those somewhere instead of just showing and loosing them afterwards

File: src/config.py (hunk 1 above)
Comment (moritzschaefer) [reply src/config.py:26]: where a pixel is text or not with the given probability from text recognition.

File: src/feature_extraction.py (hunk 2 above)
Comment (moritzschaefer) [reply src/feature_extraction.py:30]: it's ok :) 
you can assign values to functions

File: src/predict_test_img.py (hunk 3 above)
Comment (moritzschaefer) [reply src/predict_test_img.py:89]: I'll do that later, right now there is still something not working.. :/

**Pull request merged**